lbrynet/core/Session.py

```diff
@@ -195,30 +198,54 @@ def _try_upnp(self):
 
         log.debug("In _try_upnp")
 
-        def get_free_port(upnp, port, protocol):
-            # returns an existing mapping if it exists
+        def get_mapped_port(upnp, port, protocol):
+            # returns an existing mapped port if it exists
             mapping = upnp.getspecificportmapping(port, protocol)
-            if not mapping:
-                return port
-            if upnp.lanaddr == mapping[0]:
-                return mapping
-            return get_free_port(upnp, port + 1, protocol)
```

Comment:
`get_mapped_port` doesn't handle there being a redirect for a different address in the lan (it will return None)